pkg/operators/process/process.go

```
@@ -0,0 +1,642 @@
+	if instance.enabledFields[fieldCPUUsage] || instance.enabledFields[fieldCPUUsageRelative] {
+		instance.lastCPUTimes = make(map[int]uint64)
+		instance.lastSampleTime = time.Now()
+	}
+
+	return instance, nil
+}
+
+func (p *processOperator) Priority() int {
+	return Priority
+}
+
+type processOperatorInstance struct {
+	interval            time.Duration
+	dataSource          datasource.DataSource
+	done                chan struct{}
+	wg                  sync.WaitGroup
+	enabledFields       map[string]bool
+	pidField            datasource.FieldAccessor
+	ppidField           datasource.FieldAccessor
+	commField           datasource.FieldAccessor
+	cpuField            datasource.FieldAccessor
+	cpuRelativeField    datasource.FieldAccessor
+	memoryRSSField      datasource.FieldAccessor
+	memoryVirtualField  datasource.FieldAccessor
+	memoryRelativeField datasource.FieldAccessor
+	threadCountField    datasource.FieldAccessor
+	stateField          datasource.FieldAccessor
+	uidField            datasource.FieldAccessor
+	startTimeField      datasource.FieldAccessor
+	startTimeStrField   datasource.FieldAccessor
+	mountNsIDField      datasource.FieldAccessor
+	// For relative memory
+	totalMemory uint64
+	// For CPU usage calculation
+	lastCPUTimes   map[int]uint64
+	lastSampleTime time.Time
+	numCPU         int
+	// System boot time
+	bootTime time.Time
+}
+
+func (p *processOperatorInstance) Name() string {
+	return Name
+}
+
+func (p *processOperatorInstance) Start(gadgetCtx operators.GadgetContext) error {
+	// Get the system boot time
+	var bootTime time.Time
+
+	bootTime = getBootTime(gadgetCtx)
+
+	if bootTime.IsZero() {
+		gadgetCtx.Logger().Warn("Could not determine system boot time, using current time")
+		bootTime = time.Now()
+	}
+
+	p.bootTime = bootTime
+
+	// Start the process monitoring goroutine
+	p.wg.Add(1)
+	go p.monitorProcesses(gadgetCtx)
+
+	return nil
+}
+
+func getBootTime(gadgetCtx operators.GadgetContext) time.Time {
```

nit: perhaps better to return an error too and log in the caller.